src/app/modules/form/admin-form/admin-form.utils.ts

```diff
@@ -182,3 +187,35 @@ export const getAssertPermissionFn = (level: PermissionLevel): AssertFormFn => {
       return assertUnreachable(level)
   }
 }
+
+/**
+ * Reshapes given duplicate params into override props.
+ * @param params the parameters to reshape
+ * @param newAdminId the new admin id to inject
+ *
+ * @returns override props for use in duplicating a form
+ */
+export const processDuplicateOverrideProps = (
+  params: DuplicateFormBody,
+  newAdminId: string,
+): OverrideProps => {
+  const { responseMode, title } = params
+
+  const overrideProps: OverrideProps = {
+    responseMode,
+    title,
+    isNew: true,
+    admin: newAdminId,
+  }
+
+  switch (params.responseMode) {
+    case ResponseMode.Encrypt:
+      overrideProps.publicKey = params.publicKey
+      break
+    case ResponseMode.Email:
+      overrideProps.emails = params.emails
+      break
```

Comment:
need to add in an unreachable code error here?

Reply:
no need, because the params.responseMode is strongly typed to only be either ResponseMode.Email or ResponseMode.Encrypt